res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,773 @@
+            previousValue[storeIndex] = value;
+        }
+    }
+};
+
+ReloopBeatmix24.FX6Turn = function(channel, control, value, status, group) {
+    if (FxMode == 1) {
+        engine.setParameter(group, "parameter6", script.absoluteLin(value,
+            0, 1));
+    } else {
+        var effectUnit = parseInt(group.substr(23, 1), 10);
+        var storeIndex = "FX6U" + effectUnit.toString();
+        if (storeIndex in previousValue) {
+            if (value - previousValue[storeIndex] > 5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect3]", "next_effect", 1);
+                previousValue[storeIndex] = value;
+            } else if (value - previousValue[storeIndex] < -5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect3]", "prev_effect", 1);
+                previousValue[storeIndex] = value;
+            }
+        } else {
+            previousValue[storeIndex] = value;
+        }
+    }
+};
+
+ReloopBeatmix24.FX1Off = function(channel, control, value, status, group) {
+    if (FxMode != 1) {
+        engine.setValue(group, "enabled", value ? 1 : 0);
+    }
+    // Nothing in single-effect mode
+};
+
+ReloopBeatmix24.FX2Off = function(channel, control, value, status, group) {
+    if (FxMode != 1) {
+        engine.setValue(group, "enabled", value ? 1 : 0);
+    }
+    // Nothing in single-effect mode
+};
+
+ReloopBeatmix24.FX3Off = function(channel, control, value, status, group) {
+    if (FxMode != 1) {
+        engine.setValue(group, "enabled", value ? 1 : 0);
+    }
+    // Nothing in single-effect mode
+};
+
+ReloopBeatmix24.EffectClearTimerCallBack = function(group) {
+    engine.setValue(group, "clear", 0);
+};
+
+ReloopBeatmix24.FX4Off = function(channel, control, value, status, group) {
+    if (FxMode != 1) {
+        if (value == UP) {
+            engine.setValue(group, "clear", 1);
+            engine.beginTimer(100,
+                "ReloopBeatmix24.EffectClearTimerCallBack(\"" + group +
+                "\")", true);
+        }
+    }
+    // Nothing in single-effect mode
+};
+
+ReloopBeatmix24.FX5Off = function(channel, control, value, status, group) {
+    if (FxMode != 1) {
+        if (value == UP) {
+            engine.setValue(group, "clear", 1);
+            engine.beginTimer(100,
+                "ReloopBeatmix24.EffectClearTimerCallBack(\"" + group +
+                "\")", true);
+        }
+    }
+    // Nothing in single-effect mode
+};
+
+ReloopBeatmix24.FX6Off = function(channel, control, value, status, group) {
+    if (FxMode != 1) {
+        if (value == UP) {
+            engine.setValue(group, "clear", 1);
+            engine.beginTimer(100,
+                "ReloopBeatmix24.EffectClearTimerCallBack(\"" + group +
+                "\")", true);
+        }
+    }
+    // Nothing in single-effect mode
+};
+
+ReloopBeatmix24.EncoderTurn = function(channel, control, value, status, group) {
```

Comment:
Could you rename this to EffectsEncoderTurn for clarity? At first glance, I wasn't sure if this function handled the Trax encoder or the encoder in the effects section.